### PR TITLE
Use 'kernel-default-debuginfo' on SLE15-SP2+

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
Since SLE15-SP2/Leap 15.2 (standard and JeOS) there are specific kernel
packages with customized modules list.
On JeOS and some standard installation the kernel package is
'kernel-default-base', but the debuginfo package is
'kernel-default-debuginfo'.

- Related feature: https://jira.suse.com/browse/SLE-7691
- Related QA ticket: https://progress.opensuse.org/issues/56423

### CODE REVIEW

**Comment by SergioAtSUSE** [comment]
@Zaoliang, I hope you don't mind. I have gathered more information from JeOS team and Fabian. Hopefully all cases are handled with this PR.

If this PR gets accepted, it will superseed #8767

File: lib/kdump_utils.pm [hunk 1]
```diff
@@ -27,10 +27,18 @@ our @EXPORT = qw(install_kernel_debuginfo prepare_for_kdump
...
-    # JeOS uses kernel-default-base, except on aarch64 openSUSE
-    my $kernel    = script_output('rpm -qf --qf %{name} /boot/initrd-$(uname -r)');
+    my $kernel = script_output('rpm -qf --qf %{name} /boot/initrd-$(uname -r)');
     my $debuginfo = script_output('rpmquery --queryformat="%{NAME}-%{VERSION}-%{RELEASE}\n" ' . $kernel . '| sort --version-sort | tail -n 1');
     $debuginfo =~ s/$kernel/$kernel-debuginfo/g;
+    # Since SLE15-SP2+/Leap 15.2+ (standard and JeOS) there is 'kernel-default-base' but no 'kernel-default-base-debuginfo'
+    # use 'kernel-default-debuginfo' instead.
+    if ((is_sle('>=15-sp2') || is_leap('>=15.2') || is_tumbleweed) && $kernel eq 'kernel-default-base') {
+        $debuginfo =~ s/-base//g;
+        # kernel-default-base and kernel-default packages shared source and binaries, but can be created independently.
```
Comment (Vogtinator) [inline lib/kdump_utils.pm]: I don't really get this sentence, what about:

> kernel-default-base repackages kernel-default as an independent package now.

File: lib/kdump_utils.pm [hunk 2]
```diff
@@ -27,10 +27,18 @@ our @EXPORT = qw(install_kernel_debuginfo prepare_for_kdump
...
-    # JeOS uses kernel-default-base, except on aarch64 openSUSE
     my $kernel    = script_output('rpm -qf --qf %{name} /boot/initrd-$(uname -r)');
     my $debuginfo = script_output('rpmquery --queryformat="%{NAME}-%{VERSION}-%{RELEASE}\n" ' . $kernel . '| sort --version-sort | tail -n 1');
     $debuginfo =~ s/$kernel/$kernel-debuginfo/g;
+    # Since SLE15-SP2+/Leap 15.2+ (standard and JeOS) there is 'kernel-default-base' but no 'kernel-default-base-debuginfo'
+    # use 'kernel-default-debuginfo' instead.
+    if ((is_sle('>=15-sp2') || is_leap('>=15.2') || is_tumbleweed) && $kernel eq 'kernel-default-base') {
+        $debuginfo =~ s/-base//g;
+        # kernel-default-base and kernel-default packages shared source and binaries, but can be created independently.
+        # They both share kernel-default-debuginfo package. This causes kernel-default-base to have extra numbers added to the release version.
```
Comment (Vogtinator) [inline lib/kdump_utils.pm]: The "extra numbers" have nothing to do with debuginfo.

File: lib/kdump_utils.pm [hunk 3]
```diff
@@ -27,10 +27,18 @@ our @EXPORT = qw(install_kernel_debuginfo prepare_for_kdump
...
+        # Ignore the extra numbers added to the release version.
```
Comment (Vogtinator) [inline lib/kdump_utils.pm]: Add an example here

File: lib/kdump_utils.pm [hunk 4]
```diff
@@ -27,10 +27,18 @@ our @EXPORT = qw(install_kernel_debuginfo prepare_for_kdump
 
 sub install_kernel_debuginfo {
     zypper_call 'ref';
-    # JeOS uses kernel-default-base, except on aarch64 openSUSE
-    my $kernel    = script_output('rpm -qf --qf %{name} /boot/initrd-$(uname -r)');
+    my $kernel = script_output('rpm -qf --qf %{name} /boot/initrd-$(uname -r)');
```
[the remark this answers is not in the document]
Comment (SergioAtSUSE) [reply lib/kdump_utils.pm]: Yes, fixed.

File: lib/kdump_utils.pm (hunk 2 above)
Comment (SergioAtSUSE) [reply lib/kdump_utils.pm]: Yes, wrong wording. Let me try again.

**Comment by SergioAtSUSE** [comment]
## Verification runs:
- [JeOS Tumbleweed](http://openqa.slindomansilla-vm.qa.suse.de/tests/1781#step/kdump_and_crash/18)
- JeOS Leap 15.2: (coming soon)
- JeOS SLE 15-SP2
- Tumbleweed:
- [SLE 15-SP2 Build 72.1](http://openqa.slindomansilla-vm.qa.suse.de/tests/1779)
- Leap 15.2:
- SLE 12-SP5:

**Comment by Zaoliang** [comment]
for Tumbleweed-JeOS-for-kvm-and-xen-x86_64 looks good, tested with this PR:
http://f40.suse.de/tests/5261#step/kdump_and_crash/25

**Comment by SergioAtSUSE** [comment]
@Zaoliang, @Vogtinator, @foursixnine, please find verification runs in comment https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8818#issuecomment-548453395
